### ISSUE TITLE
RTMP publish stop notification

### ISSUE THREAD

**Issue by snavruzov** [ossrs/srs#585]
Hi, thanks for an amazing product,
Finally I could build SRS on my Raspberry PI and it works perfectly!
However, I have tried to find any mentions about idle publish drops, sort of nginx-rtmp feature _drop_idle_publisher_ but couldn't find any. What I'm trying is to notify clients that livestream is stopped and drop a connection between a client player and server, currently the client player is hung on the last frame and HTTP_API server http://192.168.2.111:1985/api/v1/clients/ shows that livestream is still played by the client player.
```json{
code: 0,
server: 2676,
clients: [
{
id: 149,
vhost: 2677,
stream: 2678,
ip: "192.168.2.120",
pageUrl: "",
swfUrl: "",
tcUrl: "rtmp://192.168.2.111:1935/live",
url: "/live/stream",
type: "Play",
publish: false,
alive: 3338733
}
]
}
```    

Is it possible to drop the client in this scenario? maybe I missed this option in the wiki.
Thanks

**Comment by winlinvip** [comment]
For RTMP, it's ok to play a `empty` stream, which without any audio/video data from publisher. For example, you can start SRS and play a stream, it's ok for RTMP player or server, this is `standby` mode.

It's useful for `standby` mode to support republish or encoder error. When a encoder failed or need to adjust some params, the encoder may stop then restart to publish, in this situation all clients are ok for `standby` mode.

**Comment by winlinvip** [comment]
Without `standby` mode, all clients will be disconnected when encoder republish.

So, it's ok to play a `empty` stream for RTMP, but when you want to disconnect some client, SRS provides HTTP API to kickoff a client(both player or publisher), please read https://github.com/ossrs/srs/wiki/v2_EN_HTTPApi#kickoff-client

**Comment by idxa** [comment]
Hello winlin!

I have encountered a similar problem. When the video stream ends, I need a way to notify SRS to disconnect the client. Currently, SRS is in standby mode, so even if the connection between the publisher and SRS is disconnected, the client's connection still remains. It would be very troublesome to use the kickoff interface to close the connections one by one. Additionally, I have noticed that even after both the publisher and the client are disconnected, the corresponding stream still persists and cannot be released. In scenarios with a large number of dynamic publishers, if the stream cannot be released, it may have some impact on the system's performance.

Is it possible to provide a mode similar to nginx-rtmp, where there is a no-standby mode on top of the standby mode, supporting more application scenarios? Also, is there an instruction like drop_idle_publisher to automatically detect and release streams?

Thank you!

`TRANS_BY_GPT3`
